src/modules/atommodel/particles/tokens/RadixFixedSupplyTokenDefinitionParticle.ts

```diff
@@ -0,0 +1,98 @@
+import BN from 'bn.js'
+
+import { Decimal } from 'decimal.js'
+
+import {
+    includeDSON,
+    includeJSON,
+    RadixParticle,
+    RadixSerializer,
+    RadixAddress,
+    RadixUInt256,
+    RadixOwnable,
+    RRI,
+} from '../..'
+
+/**
+ * Particle defining a fixed supply token
+ */
+@RadixSerializer.registerClass('radix.particles.fixed_supply_token_definition')
+export class RadixFixedSupplyTokenDefinitionParticle extends RadixParticle implements RadixOwnable {
+
+    
+
+    @includeDSON
+    @includeJSON
+    public rri: RRI
+    
+    @includeDSON
+    @includeJSON
+    public name: string
+
+    @includeDSON
+    @includeJSON
+    public description: string
+
+    @includeDSON
+    @includeJSON
+    public supply: RadixUInt256
+
+    @includeDSON
+    @includeJSON
+    public granularity: RadixUInt256
+
+    @includeDSON
+    @includeJSON
+    public iconUrl: string
+
+    constructor(
+        address: RadixAddress,
+        name: string,
+        symbol: string,
+        description: string,
+        supply: BN,
+        granularity: BN,
+        iconUrl: string,
+    ) {
+        super()
+
+        this.rri = new RRI(address, symbol)
+        this.name = name
+        this.description = description
+        this.supply = new RadixUInt256(supply)
```

Comment:
It looks like this constructor accepts non-positive numbers, which should not be permitted. We must have a positive (>0) supply for FixedSupplyTokenDef, so maybe add:

```javascript
 if (supply <= 0) {
     throw new Error('Fixed supply must be positive')
 }
```

*Edit:* just saw that you protect against non-positive supply in `createTokenSingleIssuance ` of `TransactionBuilder`, but why not do it here, in one place instead of many? :) 

Or better yet, create a specific DTO named `PositiveSupply`, which ensures `0 < supply <= UInt256.max`

Reply:
I'll think a little about how I want to split the error checks

Reply:
Thx for the comment, I understand your point :)